src/telliot_feed_examples/sources/morphware.py

```diff
@@ -0,0 +1,93 @@
+import json
+from dataclasses import dataclass
+from typing import Any
+from typing import Optional
+
+import requests
+from requests import JSONDecodeError
+from requests import Response
+from requests.adapters import HTTPAdapter
+from telliot_core.datasource import DataSource
+from telliot_core.dtypes.datapoint import DataPoint
+from telliot_core.dtypes.datapoint import datetime_now_utc
+from urllib3.util import Retry
+
+from telliot_feed_examples.utils.log import get_logger
+
+
+logger = get_logger(__name__)
+retry_strategy = Retry(
+    total=3,
+    backoff_factor=1,
+    status_forcelist=[429, 500, 502, 503, 504],
+    allowed_methods=["POST"],
+)
+adapter = HTTPAdapter(max_retries=retry_strategy)
+
+
+@dataclass
+class MorphwareV1Source(DataSource[str]):
+    """DataSource for Morphware query V1 expected response data."""
+
+    async def get_metadata(self) -> Optional[Response]:
+        """Returns a list of metadata strings."""
+
+        with requests.Session() as s:
+            # s.mount("https://", adapter)
+            s.mount("http://", adapter)
+            json_data = {
+                "provider": "amazon",
+                "service": "compute",
+                "region": "us-east-1",
+            }
+            try:
+                return s.post(
+                    "http://167.172.239.133:5000/products-2",
+                    headers={},
+                    json=json_data,
+                    timeout=0.5,
+                )
+
+            except requests.exceptions.RequestException as e:
+                logger.error(f"Morphware V1 API error: {e}")
+                return None
+
+    def adjust_data_types(self, data: list[str]) -> list[dict[str, Any]]:
+        return [json.dumps(d) for d in data]
+
+    async def fetch_new_datapoint(self) -> Optional[DataPoint[float]]:
+        """Retrieves Amazon EC2 instance pricing metadata from API
+        hosted by Morphware.
+
+        Returns:
+            array of JSON object strings containing EC2 metadata:
+            Interface Ec2MetaData {
+                        "Instance Type": str,
+                        "CUDA Cores": int,
+                        "Number of CPUs": int,
+                        "RAM": float,
+                        "On-demand Price per Hour": float,
+                    }
+        """
+        rsp = await self.get_metadata()
+        if rsp is None:
+            logger.warning("No response from Morphware V1 API")
+            return None, None
+
+        try:
+            ec2_metadata = rsp.json()
+        except JSONDecodeError as e:
+            logger.error("Morphware V1 source returned invalid JSON:", e.strerror)
```

Comment:
When there is error/warning logging for whatever situation. Is there a way this is reported back to Tellor? So e.g. if the endpoint goes down (say due to hardware issues), what would be the workflow that can let us know this? Or would it be expected that Morphware be constantly monitoring our endpoints?

Reply:
Ah, good question! So, since anyone can report and we don't know who our reporters are, we wouldn't know if their data sources (like this endpoint you've provided) aren't working unless we run it ourselves. But we do run them, so we will know if it goes down. We'd let you know.

But since both our teams are striving for a decentralized structure, you should run the tests or some kind of monitoring yourselves, so you aren't reliant on us. Also, this is why more data sources should be added. And ideally, sources that aren't built and maintained by the Morphware team. But idk how many sources there are for EC2 metadata 🤷

Reply:
If there is no other sources available, a good solution is for y'all to document in your data spec for this Morphware query version how others can reproduce the data (run and parse data from a scraper, for example) is your endpoint goes down 📃

Reply:
Opened an issue for this [here](https://github.com/morphware/daemon/issues/182)